scripts/repo-tool

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+################################################################################
+#      This file is part of LibreELEC - http://www.libreelec.tv
+#      Copyright (C) 2009-2016 Lukas Rusak (lrusak@libreelec.tv)
+#
+#  LibreELEC is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  LibreELEC is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+. config/options
+
+update_addons_xml() {
+  echo "[*] cleanup addons ..."
+  olddir=""
+  find target/addons/$ADDON_VERSION -iname 'changelog*.txt' | sort -rV | while read line ; do
+    dir=$(dirname $line)
+    if [ "$olddir" = "$dir" ] ; then
+      rm -f $line
+    fi
+    olddir=$dir
+  done
+
+  olddir=""
+  find target/addons/$ADDON_VERSION -iname '*.zip' | sort -rV | while read line ; do
+    dir=$(dirname $line)
+    if [ "$olddir" = "$dir" ] ; then
+      rm -f $line
+    fi
+    olddir=$dir
+  done
+
+  echo "[*] updating addons.xml* ..."
+  rm -rf .addons
+  pwd=`pwd`
+  find target/addons/$ADDON_VERSION -iname addons.xml | while read line ; do
+    localdir=`echo $line | sed s/addons.xml//g`
+    echo " [*] updating $line..."
+    echo '<?xml version="1.0" encoding="UTF-8"?>
+<addons>
+' > $line.tmp
+    for zip in $localdir/*/*.zip ; do
+      mkdir -p ".addons/$localdir"
+      unzip $zip "*/addon.xml" -d ".addons/$localdir" &>/dev/null
+    done
+    find .addons/$localdir -iname addon.xml | grep -v resources/ | while read xml ; do
+      cat $xml | grep -v "<?" >> $line.tmp
+    done
+    echo '
+</addons>' >> $line.tmp
+    mv $line.tmp $line
+    cd $localdir
+
+    md5sum addons.xml > addons.xml.md5
+    cd $pwd
+  done
+  rm -rf .addons
+}
+
+touch_addons_xml() {
+  for PROJECT in Generic RPi2 WeTek_Play; do
```

Comment:
Rather than hard-code projects (why isn't RPi listed?), if you added a helper function:
```
get_all_projects() {
  cd projects && find . -maxdepth 1 -type d -not -path . | sed 's/\.\///' | sort
}
```
this line would become `for PROJECT in $(get_all_projects); do`, and the same again further down (line 98).